src/pages/index.js

```diff
@@ -78,24 +127,128 @@ addcreateCardButton.addEventListener("click", () => {
   formValidators["add-button-form"].resetValidation();
 });
 
+avatarImage.addEventListener("click", () => {
+  avatarEditModal.open();
+});
+
 // EVENT HANDLERS
 
+function handleLikeCard(cardId) {
+  return api
+    .likeCard(cardId)
+    .then(() => {
+      const card = cardsSection.getItem(cardId);
+      if (card) {
+        card.updateLikeStatus(true);
+      }
+    })
+    .catch((err) => {
+      console.error(`Error liking card: ${err}`);
+    });
+}
+
+function handleUnlikeCard(cardId) {
+  return api
+    .unlikeCard(cardId)
+    .then(() => {
+      const card = cardsSection.getItem(cardId);
+      if (card) {
+        card.updateLikeStatus(false);
+      }
+    })
+    .catch((err) => {
+      console.error(`Error unliking card: ${err}`);
+    });
+}
+
+function handleDeleteButtonClick(cardId) {
+  deleteCardModal.open(cardId);
+}
+
 function handleProfileEditSubmit(inputValues) {
-  userInfo.setUserInfo({
+  profileSaveButton.textContent = "Saving...";
+
+  const updatedUserInfo = {
     name: inputValues.name,
-    job: inputValues.job,
-  });
+    about: inputValues.about,
+  };
 
-  profileModal.close();
+  api
+    .updateUserInfo(updatedUserInfo)
+    .then((updatedData) => {
+      userInfo.setUserInfo(updatedData);
+      profileModal.close();
+    })
+    .catch((err) => {
+      console.error("Error updating user information:", err);
+    })
+    .finally(() => {
+      profileSaveButton.textContent = "Save";
+    });
 }
 
 function handleAddCardSubmit(inputValues) {
+  addCardSaveButton.textContent = "Saving...";
   const name = inputValues.title;
   const link = inputValues.image;
   const data = { name, link };
 
-  cardsSection.addItem(createCard(data));
-  cardModal.close();
+  api
+    .createCard(data)
+    .then((newCard) => {
+      cardsSection.addItem(createCard(newCard));
+      cardModal.close();
+    })
+    .catch((err) => {
+      console.error("Error creating card:", err);
+    })
+    .finally(() => {
+      addCardSaveButton.textContent = "Create";
+    });
+}
+
+function handleDeleteCard(cardId) {
+  api
+    .deleteCard(cardId)
+    .then(() => {
+      const cardElement = document.querySelector(`[data-id='${cardId}']`);
+      if (cardElement) {
+        cardElement.remove();
+      }
+    })
+    .catch((err) => {
+      console.error("Error deleting card:", err);
+    });
+}
+
+function handleAvatarImageSubmit(inputValues) {
+  if (avatarSaveButton) {
+    avatarSaveButton.textContent = "Saving...";
+  }
+
+  const avatarUrl = inputValues.avatar;
+  if (avatarUrl) {
+    const avatarImageElement = document.querySelector(".profile__avatar");
+    if (avatarImageElement) {
+      avatarImageElement.src = avatarUrl;
```

Comment:
You need to update `DOM` only in blocks `then` after a successful response from the server so that when there is an error it wouldn’t change